### ISSUE TITLE
Metrics cannot be averaged over batches

### ISSUE THREAD

**Issue by rhine3** [kitzeslab/opensoundscape#232]
Looking through the code with Miao, he realized that the metric calculation is not right: metrics can't be calculated correctly by averaging over metrics for a batch. Instead, metrics should be calculated at the end of each epoch.  I was seeing strange behavior with the metrics before, but wasn't able to figure out what the problem was. 

Coincidentally, we made some changes made to `train.py` that will make it easier to run metrics at the end of an epoch. These changes involved making a list of the scores and labels for all batches during validation and prediction. These lists were only created if `save_scores==True`; their purpose was to save the scores and the targets to the `.tar` file. 

Instead, we should create these lists no matter what the value of `save_scores` is, by unindenting the following if statements:

https://github.com/kitzeslab/opensoundscape/blob/211e5e12464cc192ea1e5c78887bce3653b40686/opensoundscape/torch/train.py#L147-L151

https://github.com/kitzeslab/opensoundscape/blob/211e5e12464cc192ea1e5c78887bce3653b40686/opensoundscape/torch/train.py#L176-L180

Then `metrics.py` will have to be modified.

**Comment by chiroptical** [comment]
The computation of metrics happens here: https://github.com/kitzeslab/opensoundscape/blob/211e5e12464cc192ea1e5c78887bce3653b40686/opensoundscape/torch/train.py#L184-L186

**Comment by chiroptical** [comment]
The metrics function is just wrong. I'll fix it now.

**Comment by rhine3** [comment]
Fixed by #235. In the course of solving this problem we had some interesting discussions about the merits of computing metrics per epoch versus computing per batch, e.g.
* computing metrics per epoch is computing the metrics of different models (since each backprop after each epoch changes the model's state dict)
* there’s noise in the calculation of the metrics per batch since the per-batch metrics don’t represent the performance on the entire dataset
